Release Changes to Main Production

t-bello7 opened the pull request:
This pull request contains the following changes to implement a responsive fronted interface for an hotel booking application
- Design and Implement pages based on design
- Implement Jest test for testing
- Add state management for booking and cancelling a reservation
- Add authenticated pages

t-bello7 committed:
- add linter configuration

alanluqman committed:
Implement navigation

alanluqman committed:
Create redux store

alanluqman committed:
Create hotels view

alanluqman committed:
Create hotel show view

alanluqman committed:
Fix eslint errors

alanluqman committed:
Remove yarn package manager

alanluqman committed:
Merge pull request #11 from t-bello7/hotels_view

Hotels view

alanluqman committed:
Create reserve form

alanluqman committed:
Merge pull request #12 from t-bello7/reserve_form

Create reserve form

alanluqman committed:
Create add new hotel form and search box

alanluqman committed:
Merge pull request #13 from t-bello7/create_new_hotel

Create add new hotel form and search box

alanluqman committed:
Create add new room form

binhussen committed:
Merge pull request #14 from t-bello7/create_new_room

Create add new room form

binhussen committed:
Add Login Page

binhussen committed:
Add Register Page

binhussen committed:
Add Navigation For register

binhussen committed:
Add Validation

binhussen committed:
Add Error Views

binhussen committed:
Add Constants

binhussen committed:
Add User Reducer and Action

binhussen committed:
Add npm Packages

alanluqman committed:
Update components

alanluqman committed:
Create test for App.js

alanluqman committed:
Create test for Home.js component

alanluqman committed:
Create test for hotels.js component

alanluqman committed:
Create test for Booknig.js component

alanluqman committed:
Merge pull request #16 from t-bello7/unit_test

Unit test

t-bello7 committed:
fix linter issues

t-bello7 committed:
Merge pull request #15 from t-bello7/user-auth

User Login and Registration

binhussen committed:
Add User Mangment

binhussen committed:
ADd User List

binhussen committed:
Add User action and nav

binhussen committed:
update registration

binhussen committed:
Update the package

alanluqman committed:
Fix dependencies error

binhussen committed:
Merge pull request #17 from t-bello7/user-mangment

User Management

t-bello7 committed:
update readme

alanluqman committed:
install dependency

alanluqman committed:
Update home and login form design

alanluqman committed:
Update Signup page style

alanluqman committed:
Merge pull request #18 from t-bello7/home_page

Home page

t-bello7 committed:
- refactor project structure
- create layout for unauthenticated pages

t-bello7 committed:
- add style for booked rooms

t-bello7 committed:
- add style for booked rooms

t-bello7 committed:
- add rtk query for handling state management
- add check for protected pages
- refactor redux files

t-bello7 committed:
- create api request for bookings, hotels and rooms
- render data from api

t-bello7 committed:
Merge pull request #19 from t-bello7/refac-style

Refactor Styling

t-bello7 committed:
Merge pull request #20 from t-bello7/refac-redux

Refactor State mangement

Meltrust commented on README.md:
```diff
@@ -1,41 +1,12 @@
--->
-
-<div align="center">
+<!-- <div align="center"> -->
```
- [x]  On the "License" section of your readme, the link is broken because it points to a file that does not exist in your project.

### Tips and extra help

Go [here](https://github.com/microverseinc/readme-template), grab the `MIT.md` file from that repo, paste it on the root of your project, and link to it in the "License" section of your readme.

- [x] In the same license section, the note `NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish.` is just a placeholder with instructions for you, so you can add a license.  Kindly remove it after you add your license.

- [x] There are no setup instructions about how to run your project properly.  Kindly add those.  Please, don't forget to include the required commands.  (I couldn't run your setup or run the project with `npm install` and `npm start`, I had to use `yarn install, yarn start`.  Please be sure to add the instructions).  

- [x] Most of your readme is only placeholders from the Microverse readme template.  Kindly add the content for all the required sections, with proper info about your project like features, description, etc. 

- [x] Kindly,  in the "authors" section, add the correct info on the authors, with the correct links to their social media profiles, so they can be contacted.

That way, your project will be well-documented 👍

Meltrust commented on src/components/roomCard.js:
```diff
@@ -0,0 +1,141 @@
+    room_id: room.id,
+    user_id: jwt(token).user_id
+  });
+  console.log(hotelId);
```
- [x] Kindly, remove this `console.log` debugging code in order to improve the repo's cleanliness, and to meet the general [Microverse  Javascript best practices guidelines](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md). 👍

Meltrust commented on src/App.js:
```diff
@@ -1,24 +1,37 @@
 import './App.css';
+import NotFound from './components/notfound';
 
 function App() {
```
- [x]  Please, kindly refactor this and all other functions in your project to ES6 arrow functions, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered non-best practice.

You can do a search in Vscode for the "function" word, and correct as needed.  Shouldn't take you more than 5-10 min so don't worry:

![image](https://user-images.githubusercontent.com/57421823/215268988-c2498ec0-662d-40e6-8231-944e60abe26a.png)

Meltrust replied:
- [x] Please, kindly remove the old pieces of commented code found here on `App.;js`, in order to improve the repo's cleanliness. 👍

Meltrust replied:
- [ ] When I try to add a booking, I immediately get logged out of the app.  Kindly, correct this behavior so your app works as intended, and the UX is improved. 👍

Meltrust replied:
- [ ] Good job adding the "Add hotel" feature, however, I couldn't find a way to delete rooms and/or hotels.  According to the project requirements, this is a required feature:

![image](https://user-images.githubusercontent.com/57421823/215270480-7a5659b1-a292-49de-8e56-8b26439a4c72.png)

Kindly add the feature, so we can meet the project requirements.  If due to time constraints, or any unforeseeable reason you won't be able to add it, kindly include the proper explanation in your PR description.

Meltrust replied:
- [ ] When adding rooms, some fields of the form it is not possible to know what they are for.  Kindly add labels and/or placeholders so your users know what to include on each of those.  That way, the UX will be vastly improved. 👍 

![image](https://user-images.githubusercontent.com/57421823/215270970-f57cee47-2b24-4b7b-bf87-19e7f7f63738.png)

Meltrust commented on src/test/App.test.js:
```diff
@@ -0,0 +1,17 @@
+import { render, screen } from "@testing-library/react";
```
- [ ] Good job adding tests!  However, 6 of them are not passing on my end:

![image](https://user-images.githubusercontent.com/57421823/215269379-b0e5d679-d5a1-4210-ae26-3cc4a1deb3fe.png)

If they are not passing for you, kindly debug them.  If they work in that state, kindly add instructions on your readme about how to properly run them, so they pass.

Also, be sure to include a "Tests" section on your readme with instructions about how to run them (I used `yarn test`),  💪

Meltrust commented on src/components/UserList.js:
```diff
@@ -0,0 +1,114 @@
+// /* eslint-disable */
```
- [ ] This file does not contain anything but commented code.  Kindly remove this file if not using it. 👍

Meltrust commented on src/pages/User.js:
```diff
@@ -0,0 +1,64 @@
+// /* eslint-disable */
```
- [ ] This file does not contain anything but commented code.  Kindly remove this file if not using it. 👍

Meltrust commented on src/App.css:
```diff
@@ -1,38 +1,119 @@
-.App {
+:root {
```
- [ ] Some pages in the mobile view are not very responsive:

![image](https://user-images.githubusercontent.com/57421823/215269880-b4fc05d0-d6e0-471e-9893-4e1c6ccbcfc9.png)

Kindly work a little more on your design, so it looks more professional and the UX is improved. 👍

Meltrust replied:
- [ ] All of the images are not showing up.  You could use example images found on the internet (link to them, so you don't have to use active storage if you don't want to).  

## Tips and extra help

_Steps:_

### In the backend

- Update the models to include the image_url field.
- Add the proper image_url fields to the tables in the database via migrations.
- Update controllers so the params and actions include the image_url, that way, your API will render it in your JSON.

### In the frontend

- Include the images on your JSON fetches to the API.
- Update the redux action creators and reducers.
- Update the JSX in the proper components to include the images.
- Profit.


That way, your project will look beautiful, and professional. 💪

Meltrust commented on src/components/booking.js:
```diff
@@ -0,0 +1,61 @@
+import { useGetBookingsQuery } from '../services/hotel';
+import Loader from './Loader';
+
+export default function Booking() {
```
- [ ] The "Bookings" table is not very informative:

![image](https://user-images.githubusercontent.com/57421823/215270238-132a1529-5c8b-49d3-a718-73df4e03a12d.png)

Kindly, add more info (like the dates, and the room/hotel in question). 👍

Meltrust replied:
- [ ] Good job adding a search bar!  However, in the "Bookings" section, the moment I try to search for anything I get a white screen.  Kindly fix this bug so your app works as intended. 💪

Meltrust commented on src/components/hotelCard.js:
```diff
@@ -0,0 +1,47 @@
+import { NavLink } from 'react-router-dom';
+import "../assets/styles/hotelCard.css";
+
+export default function HotelCard(props) {
```
- [x] In the hotel cards, there is a field for "Size", but it contains no info.  Kindly either add the correct "size" info or remove the field.  That way, we ensure consistency. 👍 

![image](https://user-images.githubusercontent.com/57421823/215270846-06cb3dbb-de38-4308-a144-f44ad3941ec7.png)

Meltrust reviewed (changes requested):
Hi @t-bello7,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  2 tests are passing✔️
-  Nice start ✔️ 
-  Good layout ✔️ 
-  Design colors match ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

![You can do it](https://media.giphy.com/media/FemTGpEZeK27M5fKL4/giphy.gif)

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._

_I strongly recommend you take them into account as they can improve your final evaluation._





Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

binhussen committed:
Update Readme

binhussen committed:
Clear Unused File

binhussen committed:
Add License

binhussen committed:
Clear Comments

binhussen committed:
Update Data

binhussen committed:
Update to es6 feature

binhussen committed:
- fix linter errors

alanluqman committed:
Update booking view

t-bello7 committed:
Merge pull request #22 from t-bello7/re-arrange

Re arrange

t-bello7 committed:
Merge branch 'booking_refactory' into re-arrange

t-bello7 committed:
Merge pull request #23 from t-bello7/booking_refactory

Update booking view

alanluqman committed:
Make website responsive to mobile

t-bello7 committed:
Merge changes from remote

t-bello7 committed:
Merge pull request #24 from t-bello7/mobile_responsive

Make website responsive to mobile

alanluqman committed:
Make booking tab responsive

t-bello7 committed:
Merge pull request #25 from t-bello7/booking_tab__responsive

Make booking tab responsive

t-bello7 committed:
- add token to localstorage
- get token form localstorage to update auth state
- add tbody and thead to provide structure for table

t-bello7 committed:
Merge pull request #26 from t-bello7/localstorage

Add Local Storage Implementation

binhussen committed:
Implemet Hhotel Update

t-bello7 committed:
Merge pull request #27 from t-bello7/hotel-update

Implemet Hotel Update

binhussen committed:
Add Hotel Delete Functionality

binhussen committed:
Add query for edit and delete room

binhussen committed:
Add Edit and Delete functionality for room

binhussen committed:
Add Loders

t-bello7 committed:
- add redux delete action to hotels page
- add redux update action to room to update the reserved state
- update routes
- add default image for hotels and rooms

t-bello7 committed:
- add redux delete action to hotels page
- add redux update action to room to update the reserved state
- update routes
- add default image for hotels and rooms

t-bello7 committed:
Merge pull request #28 from t-bello7/room-update

Room update

t-bello7 committed:
Merge remote dev branch

t-bello7 committed:
- add image input to update form for hotel and room

t-bello7 committed:
Merge pull request #29 from t-bello7/update-delete

Update and Delete Dispatch

t-bello7 committed:
fix login

Meltrust commented on src/App.js:
```diff
@@ -1,25 +1,33 @@
-    </div>
-  );
-}
+const App = () => (
```
- [ ] When I try to register either as an admin or a regular user, I get greeted by a white screen and these console errors:  This happens locally and in the live demo.

![image](https://user-images.githubusercontent.com/57421823/216627155-37499ef1-b21d-47f7-80ec-eff567a50013.png)

After taking a look at dev-tools, It seems this happens if the user registration fails (server response: email has already been taken).  Kindly, if the registration fails inform the user what just happening and prevent them to be greeted by a white screen if their registration fails.

![image](https://user-images.githubusercontent.com/57421823/216629874-7e14bea5-cf52-4aaa-bfdd-f7cb2446c5ec.png)

Meltrust replied:
- [ ] It seems that the only hotels that can be deleted, are those without reservations (good job doing this validation).  I noticed these because I got 500 from the server when I tried to delete a hotel that had reservations.  Now, kindly let the user know with a short message when this happens.  That way, your users will be able to correctly delete hotels and rooms and will know that in order to be able to be able to delete a hotel, the reservations should be canceled first. 👍

Meltrust commented on README.md:
```diff
@@ -255,9 +209,11 @@ If you like this project...
+- LazyCoders
+- Coding Partners
+I would like to thank Marten Bjork for the image
+https://unsplash.com/photos/n_IKQDCyrG0?utm_source=unsplash&utm_medium=referral&utm_content=creditShareLink
```
On the "License" section of your readme:

- [ ] This message:  

`_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._`

This is just a placeholder with instructions for you to add a license.  Kindly remove it (after reading it).

- [ ]  On the "License" section of your readme, the link is broken because it points to a file that does not exist in your project.

## Tips and extra help

Go [here](https://github.com/microverseinc/readme-template), grab the `MIT.md` file from that repo, paste it on the root of your project, and link to it in the "License" section of your readme.

You already did this correctly on your backend app.  Kindly add the license file here in the frontend repo and correct the link.

That way, your app will be well documented 💪

Meltrust commented on README.md:
```diff
@@ -255,9 +209,11 @@ If you like this project...
-> Give credit to everyone who inspired your codebase.
-
-I would like to thank...
+I would like to thank Microverse for giving me the opportunity to work on this project.
```
- [ ] Your frontend tests are not passing on my end:

![image](https://user-images.githubusercontent.com/57421823/216634216-b20cdfba-b0a4-4546-893e-a09f8484a31c.png)

I used `yarn test` as per your readme instructions.

Kindly, fix them so they all pass.  Otherwise, if there are special procedures to properly run them, kindly include those instructions in your readme file. 👍

Meltrust reviewed (changes requested):
_note:  As indicated on the dashboard, this is first review._

Hi @t-bello7,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:


-  App is working well (if registration succeeds)✔️ 
-  Readme has been improved✔️ 
-  Beautiful, nicely reworked frontend ✔️ 
-  Empty files have been removed ✔️
-  All functions are ES6  ✔️
-  No console.log statements and/or alerts were found ✔️
-  Nice mobile view ✔️ 


#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

![You can do it](https://media.giphy.com/media/FemTGpEZeK27M5fKL4/giphy.gif)

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._

_I strongly recommend you take them into account as they can improve your final evaluation._





Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------
